Issue291 - define external sites (#322)
redhat-documentation/supplementary-style-guide: supplementary_style_guide/style_guidelines/links.adoc

Follow these guidelines when linking externally:

> * Always include accurate link text. Bare URLs (URLs without explanatory information) are unhelpful because they do not provide adequate context about the link target.

Reviewer: "Always include link text, and ensure that the link text is descriptive. Bare URLs (URLs without explanatory information) and generic link text (such as "click here") are unhelpful because they do not provide adequate context about the link target."

> * Always include accurate link text. Bare URLs (URLs without explanatory information) are unhelpful because they do not provide adequate context about the link target.

Reviewer: I think your point about avoiding generic link text is good and should be included too, preferably in the second sentence.

* Use hyperlinks unless the URL is an example URL or is otherwise inaccessible to users. For information about using addresses in examples, see the _IBM Style_ guide.

> * By default, ensure that links are followable and crawlable. Do not use the "nofollow" link option unless absolutely necessary.

Reviewer: Sorry, I had originally made this comment before the bullet point was moved! The point I disagreed with was the addition of the text "ensure that" :)

> * By default, ensure that links are followable and crawlable. Do not use the "nofollow" link option unless absolutely necessary.

Author: Yes. Excellent point. I misunderstood the original text! Will revert.

* When possible, link to a top-level page and avoid deep links to a specific page or image.

> Linking to a top-level page helps minimize future link maintenance and can avoid inadvertent bypassing of a site's legal information.

Reviewer 2: "can require" or "can cause" instead of "incur" for non-native English speakers.

> Linking to a top-level page helps minimize future link maintenance and can avoid inadvertent bypassing of a site's legal information.

Author: Not sure I agree with this suggestion. All links, whether deep or not, require some maintenance. What I want to express is that we want to minimize link maintenance, and that avoiding deep links helps us achieve that goal.

> Linking to a top-level page helps minimize future link maintenance and can avoid inadvertent bypassing of a site's legal information.

Reviewer: "Deep links can break more frequently..."?

> Linking to a top-level page helps minimize future link maintenance and can avoid inadvertent bypassing of a site's legal information.

Author: Excellent suggestion!

> * When feasible, avoid links to external sites, which are sites other than those owned and operated by Red Hat or IBM.

Reviewer 2: ```suggestion
* Whenever possible, avoid links to external sites, which are sites other than those owned and operated by Red Hat or IBM.
```
Non-native speakers might have a hard time with "feasible".

Author: You're right: "possible" is better than "feasibel"

> * When feasible, avoid links to external sites, which are sites other than those owned and operated by Red Hat or IBM.

Author: Good point! I'll make this change.

> * When feasible, avoid links to external sites, which are sites other than those owned and operated by Red Hat or IBM.

Reviewer 2: We are supposed to avoid parentheticals in documentation.

> * When feasible, avoid links to external sites, which are sites other than those owned and operated by Red Hat or IBM.

Reviewer: I don't mind about removing the parentheses, but I would like to emphasise that external links should be avoided unless essential. I think that's stronger than "when feasible" or "when possible".

> * When feasible, avoid links to external sites, which are sites other than those owned and operated by Red Hat or IBM.

Author: What about `Avoid unnecessary links to external sites, which are sites other than those owned and operated by Red Hat or IBM.`

Follow these guidelines when linking externally:

* Avoid unnecessary links to external sites not owned and operated by Red Hat or IBM.
Links to external sites can change or be unreliable.
In addition, customers might infer that Red Hat endorses or supports the linked content, even if that is not the intent.
+
[NOTE]
====
Links to upstream sites, such as GitHub, are considered to be external links.
====
+
* When possible, link to a top-level page and avoid deep links to a specific page or image.
Deep links can break more frequently and can inadvertently bypass a site's legal notices.

> * Always include meaningful link text.

Reviewer: This point (meaningful link text) should come after the point about not using bare URLs. 

We are making 2 points here:
1. Use link text. (aka "Don't use bare URLs.")
2. Use _meaningful_ link text. [that is, don't use "click here" and other such generic text]

Author: Thanks. That makes sense.

@bredamc @sbmetz @mportman12  - Can you all please take another look? Thank you!

* Always include link text. Bare URLs (URLs without explanatory information) are unhelpful because they do not provide adequate context about the link target.
* Use hyperlinks unless the URL is an example URL or is otherwise inaccessible to users. For information about using addresses in examples, see the _IBM Style_ guide.
* By default, links are followable and crawlable. Do not use the "nofollow" link option unless absolutely necessary.

> * Avoid deep links to a specific page or image to prevent an ongoing maintenance responsibility and to prevent bypassing a website's legal information.

Reviewer 3: ```suggestion
* When possible, link to a top-level page and avoid deep links to a specific page or image. Linking to a top-level page prevents an ongoing maintenance responsibility and prevents bypassing a website's legal information.
```

* Always include link text. Bare URLs (URLs without explanatory information) are unhelpful because they do not provide adequate context about the link target.
* Use hyperlinks unless the URL is an example URL or is otherwise inaccessible to users. For information about using addresses in examples, see the _IBM Style_ guide.
* By default, links are followable and crawlable. Do not use the "nofollow" link option unless absolutely necessary.

> * Avoid deep links (to a specific page or image from another company) to prevent an ongoing maintenance responsibility and to prevent bypassing a website's legal information.

Reviewer: ```suggestion
* Avoid deep links (to a specific page or image) to prevent an ongoing maintenance responsibility and to prevent bypassing a website's legal information.
```

Reviewer: The original advice in IBM Style (about avoiding deep links) does not refer to external sites.

> * Avoid deep links (to a specific page or image from another company) to prevent an ongoing maintenance responsibility and to prevent bypassing a website's legal information.

Author: Can you explain further? I'm not quite understanding this question.

> * Avoid deep links (to a specific page or image from another company) to prevent an ongoing maintenance responsibility and to prevent bypassing a website's legal information.

Reviewer 3: "Avoid deep links to a specific page from another company" -- to me that implies it's better to link to a top-level doc portal page. If so, then do we recommend to link to _just_ the top level page, or name the topic where relevant to help readers navigate from the top page?

> * Avoid deep links (to a specific page or image from another company) to prevent an ongoing maintenance responsibility and to prevent bypassing a website's legal information.

Author: That's a good question. My own take is that we shouldn't be absolutely prescriptive. Maybe add something along the lines of "When possible, link to top-level pages and avoid deep links to a specific page or image."

> * Avoid deep links (to a specific page or image from another company) to prevent an ongoing maintenance responsibility and to prevent bypassing a website's legal information.

Reviewer 3: I made a suggestion; feel free to tweak.

> * Avoid links to external sites when possible. External links can change or be unreliable.

Reviewer: ```suggestion
* Avoid links to external sites when possible. Links to external sites can change or be unreliable, and customers might infer that Red Hat endorses the linked content, even if that is not the intent.
```

Reviewer: This bullet point (about _external sites_) is in a complete section about _external links_. We need to distinguish between the two terms.

Author: Changed per your suggestion.

> External sites are considered to be sites other than those owned and operated by Red Hat.

Reviewer: ```suggestion
External sites are sites other than those owned and operated by Red Hat and IBM.
```

Author: Changed to "or IBM". As a subsidiary Red Hat is a separate entity, even though it is wholly owned by IBM.

+
[NOTE]
====
Links to upstream documentation sites are considered to be external links.

> Linking to an upstream site does not imply that Red Hat supports or endorses any content on that site.

Reviewer 3: > Linking to an upstream site does not imply that Red Hat supports or endorses any content on that site.

This line seems out of place -- it's not really guidance about using cross-references to external links, is it?

Author: @sbmetz Yeah, I see what you mean. I was picking up the need for this from the Slack thread linked in Issue #291. Maybe it would be best to leave this out. I'd be interested in what others have to say.

Reviewer: Agree that we should delete the second note -- if we want to keep the text about inferred endorsement, we can add it in the main bullet point as I've suggested in a comment above.

Author: Removed. I also made the sentence more generic to refer to all upstream sites (not just documentation) and added the example of GitHub.
